cunumeric/array.py

```diff
@@ -2768,8 +2920,12 @@ def flatten(self, order: OrderType = "C") -> ndarray:
         Multiple GPUs, Multiple CPUs
 
         """
-        # Same as 'ravel' because cuNumeric creates a new array by 'reshape'
-        return self.reshape(-1, order=order)
+        # Reshape first and make a copy if the output is a view of the src
+        # the output always should be a copy of the src array
+        result = self.reshape(-1, order=order)
+        if result.base is not None:
```

Comment:
the `base` property is not correctly implemented. so if you wanted to use it, you should have at least checked that it does what you wanted. if you want to check whether the `result` is a view, you should add a proper check, say `is_view`, to both EagerArray and DeferredArray and use it here (i.e., `if result._thunk.is_view`), because their flattening implementations exhibit different behaviors (see below).